src/app/api/api_v1/endpoints/detections.py

```diff
@@ -89,20 +106,38 @@ async def get_detection_url(
 @router.get("/", status_code=status.HTTP_200_OK, summary="Fetch all the detections")
 async def fetch_detections(
     detections: DetectionCRUD = Depends(get_detection_crud),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> List[Detection]:
     telemetry_client.capture(token_payload.sub, event="detections-fetch")
-    return [elt for elt in await detections.fetch_all()]
+    all_detections = [elt for elt in await detections.fetch_all()]
+    if UserRole.ADMIN in token_payload.scopes:
+        return all_detections
+    filtered_detections = []
+    for detection in all_detections:
+        camera = await cameras.get(detection.camera_id, strict=True)
+        if camera is not None and camera.organization_id == token_payload.organization_id:
+            filtered_detections.append(detection)
+    return filtered_detections
```

Comment:
Mmmh this is why I would also argue that org shouldn't prevent read on detections. This step is quite heavy on the DB once there are a lot of detections. We're talking about switching from 1 fetch operations on N rows, to 1 fetch + N get. I don't think we can afford this

Reply:
yes I thought about it, but we could implement it differently in our crud with a GROUP BY so i thought that I would do it later.

Reply:
I can do it now

Reply:
Happy with any sql ops that avoids this heavy load 🙂